diff --git a/project-3/contracts/Migrations.sol b/project-3/contracts/Migrations.sol
index c4efb65..85050b0 100755
--- a/project-3/contracts/Migrations.sol
+++ b/project-3/contracts/Migrations.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.23;
+pragma solidity = 0.5.16;
 
 contract Migrations {
   address public owner;
diff --git a/project-3/contracts/coffeeaccesscontrol/ConsumerRole.sol b/project-3/contracts/coffeeaccesscontrol/ConsumerRole.sol
index 8eb1bd2..25b9d88 100755
--- a/project-3/contracts/coffeeaccesscontrol/ConsumerRole.sol
+++ b/project-3/contracts/coffeeaccesscontrol/ConsumerRole.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 
 // Import the library 'Roles'
 import "./Roles.sol";
diff --git a/project-3/contracts/coffeeaccesscontrol/DistributorRole.sol b/project-3/contracts/coffeeaccesscontrol/DistributorRole.sol
index 5de1ee2..f69a895 100755
--- a/project-3/contracts/coffeeaccesscontrol/DistributorRole.sol
+++ b/project-3/contracts/coffeeaccesscontrol/DistributorRole.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 
 // Import the library 'Roles'
 import "./Roles.sol";
diff --git a/project-3/contracts/coffeeaccesscontrol/FarmerRole.sol b/project-3/contracts/coffeeaccesscontrol/FarmerRole.sol
index 7d72042..5711a38 100755
--- a/project-3/contracts/coffeeaccesscontrol/FarmerRole.sol
+++ b/project-3/contracts/coffeeaccesscontrol/FarmerRole.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 
 // Import the library 'Roles'
 import "./Roles.sol";
diff --git a/project-3/contracts/coffeeaccesscontrol/RetailerRole.sol b/project-3/contracts/coffeeaccesscontrol/RetailerRole.sol
index 003252b..95c7b81 100755
--- a/project-3/contracts/coffeeaccesscontrol/RetailerRole.sol
+++ b/project-3/contracts/coffeeaccesscontrol/RetailerRole.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 
 // Import the library 'Roles'
 import "./Roles.sol";
diff --git a/project-3/contracts/coffeeaccesscontrol/Roles.sol b/project-3/contracts/coffeeaccesscontrol/Roles.sol
index a62ad28..ac1e94f 100755
--- a/project-3/contracts/coffeeaccesscontrol/Roles.sol
+++ b/project-3/contracts/coffeeaccesscontrol/Roles.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 
 /**
  * @title Roles
diff --git a/project-3/contracts/coffeebase/SupplyChain.sol b/project-3/contracts/coffeebase/SupplyChain.sol
index 782222c..35eb813 100755
--- a/project-3/contracts/coffeebase/SupplyChain.sol
+++ b/project-3/contracts/coffeebase/SupplyChain.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 // Define a contract 'Supplychain'
 contract SupplyChain {
 
@@ -49,7 +49,7 @@ contract SupplyChain {
     State   itemState;  // Product State as represented in the enum above
     address distributorID;  // Metamask-Ethereum address of the Distributor
     address retailerID; // Metamask-Ethereum address of the Retailer
-    address consumerID; // Metamask-Ethereum address of the Consumer
+    address payable consumerID; // Metamask-Ethereum address of the Consumer
   }
 
   // Define 8 events with the same 8 state values and accept 'upc' as input argument
@@ -145,15 +145,9 @@ contract SupplyChain {
     upc = 1;
   }
 
-  // Define a function 'kill' if required
-  function kill() public {
-    if (msg.sender == owner) {
-      selfdestruct(owner);
-    }
-  }
 
   // Define a function 'harvestItem' that allows a farmer to mark an item 'Harvested'
-  function harvestItem(uint _upc, address _originFarmerID, string _originFarmName, string _originFarmInformation, string  _originFarmLatitude, string  _originFarmLongitude, string  _productNotes) public 
+  function harvestItem(uint _upc, address _originFarmerID, string memory _originFarmName, string memory _originFarmInformation, string memory  _originFarmLatitude, string memory  _originFarmLongitude, string memory  _productNotes) public 
   {
     // Add the new item as part of Harvest
     
@@ -269,10 +263,10 @@ contract SupplyChain {
   uint    itemUPC,
   address ownerID,
   address originFarmerID,
-  string  originFarmName,
-  string  originFarmInformation,
-  string  originFarmLatitude,
-  string  originFarmLongitude
+  string  memory originFarmName,
+  string  memory originFarmInformation,
+  string  memory originFarmLatitude,
+  string  memory originFarmLongitude
   ) 
   {
   // Assign values to the 8 parameters
@@ -297,12 +291,12 @@ contract SupplyChain {
   uint    itemSKU,
   uint    itemUPC,
   uint    productID,
-  string  productNotes,
+  string  memory productNotes,
   uint    productPrice,
   uint    itemState,
   address distributorID,
   address retailerID,
-  address consumerID
+  address payable consumerID
   ) 
   {
     // Assign values to the 9 parameters
diff --git a/project-3/contracts/coffeecore/Ownable.sol b/project-3/contracts/coffeecore/Ownable.sol
index 73e8e31..3af6e6a 100644
--- a/project-3/contracts/coffeecore/Ownable.sol
+++ b/project-3/contracts/coffeecore/Ownable.sol
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity = 0.5.16;
 
 /// Provides basic authorization control
 contract Ownable {
